logstash-core/src/main/java/org/logstash/common/io/DeadLetterQueueReader.java

```diff
@@ -223,6 +243,32 @@ byte[] pollEntryBytes(long timeout) throws IOException, InterruptedException {
         return event;
     }
 
+    /**
+     * Acknowledge last read event, must match every {@code #pollEntry} call.
+     * */
+    public void markForDelete() {
+        if (!cleanConsumed) {
+            // ack-event is useful only when clean consumed is enabled.
+            return;
+        }
+        if (lastConsumedReader == null) {
+            // no reader to a consumed segment is present
+            return;
+        }
+
+        segmentCallback.segmentCompleted();
+
+        Path lastConsumedSegmentPath = lastConsumedReader.getPath();
+        try {
```

Comment:
Could this section of code use `delete/trashSegment`?

Reply:
Good point!

Reply:
fixed by 990976853b58b852a9f9c26ff1de4ca65d10e645